command-expansion-server/sql/commanding/tables/expansion_set_to_rule.sql

```diff
@@ -1,22 +1,27 @@
 create table expansion_set_to_rule (
   set_id integer not null,
   rule_id integer not null,
+  activity_type text not null,
 
   constraint expansion_set_to_rule_primary_key
   primary key (set_id,rule_id),
 
   foreign key (set_id)
   references expansion_set (id),
 
-  foreign key (rule_id)
-  references expansion_rule (id)
+  foreign key (rule_id, activity_type)
+  references expansion_rule (id, activity_type),
```

Comment:
I'm glad this sort of thing worked :+1: though you would think that postgres could figure out that if `id` is unique, (`(id, activity_type)`) must also be unique... 🤷

Reply:
Exactly my thoughts.